network-api/networkapi/wagtailpages/tests/buyersguide/test_homepage.py

```diff
@@ -297,3 +300,73 @@ def test_category_page_context_with_cta_enabled(self):
 
         self.assertEqual(response.status_code, 200)
         self.assertIsNotNone(response.context['featured_cta'])
+
+
+class BuyersGuideCallToActionTest(TestCase):
+
+    def setUp(self):
+        edit_handler = get_snippet_edit_handler(BuyersGuideCallToAction)
+        self.form_class = edit_handler.get_form_class()
+
+    def test_return_target_url_with_external_link(self):
+        test_url = "http://test.com"
+        cta = BuyersGuideCallToAction.objects.create(
+            title="Test CTA",
+            link_label="Test Link",
+            link_target_url=test_url,
+        )
+        cta.save()
+
+        target_url = cta.get_target_url()
+
+        self.assertEqual(target_url, test_url)
+
+    def test_return_target_url_with_page_link(self):
+        test_page = buyersguide_factories.BuyersGuidePageFactory()
+        cta = BuyersGuideCallToAction.objects.create(
+            title="Test CTA",
+            link_label="Test Link",
+            link_target_page=test_page,
+        )
+        cta.save()
+
+        target_url = cta.get_target_url()
+
+        self.assertEqual(target_url, test_page.url)
+
+    def test_cannot_be_missing_external_or_page_link(self):
+        form = self.form_class(
+            data=({"title": "Test CTA", "link_label": "Test Link"}),
+        )
```

Comment:
Can you add some context as to why you are using a form to test the model validation?

Reply:
Hey Tibor, no exact reasoning, I just used the [BuyersGuideProductCategoryTest](https://github.com/mozilla/foundation.mozilla.org/blob/57b36340228827017351b667117999973cf6a998/network-api/networkapi/wagtailpages/tests/buyersguide/test_products.py#L519) for reference as they are both snippets. Also, this made sense to me as the errors are being returned in the form errors and I was able to check for specific errors being returned. 
EX: If I add a external link and page link but no label, I want to make sure the field with the error is "link_label"